sys-apps/groff/files/groff-1.22.4-cross-compile.patch

```diff
@@ -0,0 +1,13 @@
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -4228,8 +4228,8 @@ prefixexecbin_SCRIPTS = chem neqn nroff
+ 
+ # Path to binaries and flags used by contribs and doc to generated doc.
+ # These may be overridden if cross-compiling.
+-GROFFBIN = $(abs_top_builddir)/groff
+-GROFF_BIN_PATH = $(abs_top_builddir)
++GROFFBIN = ${EPREFIX}/usr/bin/groff
```

Comment:
A few problems here:
- `EPREFIX` is a gentoo-specific variable
- the comment above explicitly says you can override those variables via `make` as `make GROFFBIN=/path/to/something`
- This file is autogenerated from Makefile.am. If you change it you should apply change there.

I think you can do an override in ebuild only.

Reply:
Were I to override the variable in the ebuild, would that not introduce groff
itself as a build-dep? I wouldn't mind it so much if EAPI=7 were available
for base-system packages but until a year or so after its introduction it can
not be used for them.

Reply:
Well, this is when you're cross-compiling, and you can't execute an aarch64 groff
binary when cross-compiling, so if it were using crossdev, you'd already have it on
the host system (since its part of `@system`), and if you're building it normally, well,
you still would already have it installed because its part of `@system`.

Reply:
But that's the problem of base-system packages: We must support bootstrapping, i.e. scenarios where you don't complete @system set yet.

Also, we don't really like conditional patching...

Reply:
And conditional sedding is fine, then? it achieves the same goal, and again, this only
happens on cross-compiles (eg crossdev sysroots) which I don't think gets bootstrapped
anyways since you can't execute the groff binary for a aarch64/arm/etc build on x86/amd64
without using qemu's user mode or the like.

Reply:
I mean, I could prolly cook something up like `app-text/scdoc` to where if its a cross
compile it would build a copy of groff for whatever the host system is (in my case,
amd64) and use that to build the groff docs for the aarch64/etc build, would take a
bit of tinkering.